smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/HttpFileUploadManager.java

```diff
@@ -494,11 +497,12 @@ private void upload(InputStream iStream, long fileSize, Slot slot, UploadProgres
             case HttpURLConnection.HTTP_NO_CONTENT:
                 break;
             default:
-                throw new IOException("Error response " + status + " from server during file upload: "
-                                + urlConnection.getResponseMessage() + ", file size: " + fileSize + ", put URL: "
-                                + putUrl);
+                throw new UnsuccessfulUploadException(status, urlConnection.getResponseMessage(), fileSize, putUrl);
             }
         }
+        catch (IOException e) {
+            throw new UnexpectedUploadException(fileSize, slot, e);
```

Comment:
As above, I would name things exception `HttpUploadIoException`, since it basically wraps an `IOException`.